lib/color-validate.spec.js

```diff
@@ -0,0 +1,154 @@
+'use strict';
+
+const { expect } = require('chai');
+const validColor = require('./color-validate.js');
+
+describe('Color validater', function () {
+  it('should check if given valid hex colors', function () {
+    const colorsTrue = [
+        '#4c1',
+        '#4C1',
+        '#abc123',
+        '#ABC123',
+        '#ABC123 ',
+      ];
+    const colorsFalse = [
+        '#123red', // contains letter above F
+        '#red', // contains letter above F
+        '123456', // contains no # symbol
+        '123', // contains no # symbol
+      ];
+    colorsTrue.forEach((color) => {
+      expect(validColor(color)).to.be.true;
+    });
+    colorsFalse.forEach((color) => {
+      expect(validColor(color)).to.be.false;
+    });
+  });
+
+  it('should check if given valid rgb colors', function () {
+    const colorsTrue = [
+        'rgb(0,109,255)',
+        'rgb(255, 0, 128)',
+        'rgb(128,255,0)',
+        'rgb(0,10, 100)',
+        'rgb(255, 255,255)',
+        'rgb(0,0,0)',
+        'rgb(0,0,0) ',
+      ];
+    const colorsFalse = [
+        'rgb(256,128,255)', // r > 255
+        'rgba(-10,128,255)', // r < 0
+        'rgb(0 ,128,255)', // space before ','
+        'rgb(0,0,280)', // g > 255
+        'rgb(05,0,255)', // r starts with 0
+      ];
+    colorsTrue.forEach((color) => {
+      expect(validColor(color)).to.be.true;
```

Comment:
If one of values fails, we can't see a a input value in logs:
```
  1) Color validater
       should check if given valid rgb colors:

      AssertionError: expected false to be true
      + expected - actual

      -false
      +true
```
This can be changed to:
```js
expect(validColor(color), color + 'is a valid color').to.be.true;
```
And the we get:
```
  1) Color validater
       should check if given valid rgb colors:

      rgb(000,0,0) is a valid color
      + expected - actual

      -false
      +true
```
Alternatively we can use https://github.com/mikec/sazerac for such tests. An example: https://github.com/badges/shields/blob/37617c1d4ed155525582bc5b4528fa1c259cee08/lib/color-formatters.spec.js#L75-L83

Reply:
Ah thanks for the info, this makes sense, I wasn't aware that's how you get the message to appear.

Will update to use: 
```javascript
expect(validColor(color), `${color} is a valid color`).to.be.true;
expect(validColor(color), `${color} is a valid color`).to.be.false;
```
and will update `make-badge.spec.js to: _(not sure of a better message)_
```javascript
expect(JSON.parse(makeBadge({ text: ['name', 'Bob'], colorscheme: color, format: 'json', template: '_shields_test' })).colorB, `${color} should return undefined`).to.be.undefined;
```

Reply:
What do you think about:
```js
expect(validColor(color), `${color} is a valid color`).to.be.true;
expect(validColor(color), `${color} is not a valid color`).to.be.false;
```
Because every value in "... to.be.true" is valid, and values in "... .to.be.false" are invalid.

Reply:
Wasn't sure about the message as `false - blu is not a valid color` almost feels like you are saying that `blu` should be a valid color.
```cmd
      blu is not a valid color
      + expected - actual

      -true
      +false
```

What about something like:
```javascript
expect(validColor(color), `is ${color} a valid color`).to.be.true;
expect(validColor(color), `is ${color} a valid color`).to.be.false;
```

Reply:
I like the first option. You're right, it looks better.

Reply:
:+1: Have updated